server/src/test/java/org/opensearch/common/time/DateFormattersTests.java

```diff
@@ -683,4 +684,17 @@ public void testCamelCaseDeprecation() {
             }
         }
     }
+
+    public void testToVerifyFixForDateTimeException() {
+        DateFormatter formatter = DateFormatter.forPattern("strict_date_optional_time||epoch_millis");
+        try {
+            formatter.parse("2018-03-30T17-30-28.842Z");//text causes DateTimeException
```

Comment:
Could you please use `assertThrows(DateTimeException.class, () -> formatter.parse("2018-03-30T17-30-28.842Z"));`

Reply:
It's actually not expected to see DateTimeException with the fix, formatter will throw IllegalArgumentException with the fix.

w/o the fix it throws DateTimeException. 

I have the verification the catch blocks, test case will fail when there is DateTimeException leaked from DateFormatter.

Reply:
Oops, sorry, shouldn't it throw `DateTimeParseException` ?

Reply:
Yes, current implementation of JavaDateFormatter doesn't respect the document on interface DateFormatter, it catches DateTimeParseException and re-throw IllegalArgumentException.

Also fixed it in new revision, please refer to the latest diff in PR.